web/src/compose_ui.js

```diff
@@ -672,6 +672,29 @@ export function format_text($textarea, type, inserted_content) {
             $textarea.caret(range.end + `[](${inserted_content})`.length);
             break;
         }
+        case "latex": {
+            const inline_latex_syntax = "$$";
+            let block_latex_syntax_start = "```math\n";
+            let block_latex_syntax_end = "\n```";
+            // If the selected text is multiline or already using multiline math syntax,
+            // we use multiline math syntax.
+            if (
+                selected_text.includes("\n") ||
+                is_selection_formatted(block_latex_syntax_start, block_latex_syntax_end)
+            ) {
+                // Add newlines before and after, if not already present.
+                if (range.start > 0 && text[range.start - 1] !== "\n") {
+                    block_latex_syntax_start = "\n" + block_latex_syntax_start;
+                }
+                if (range.end < text.length && text[range.end] !== "\n") {
+                    block_latex_syntax_end = block_latex_syntax_end + "\n";
+                }
+                format(block_latex_syntax_start, block_latex_syntax_end);
+            } else {
+                format(inline_latex_syntax);
+            }
+            break;
+        }
```

Comment:
Playing around with the LaTeX/code ones, I think it'd be better to default to the block syntax when nothing is selected and you're on a blank line; I imagine that's a modest tweak to this logic?